pyiceberg/avro/writer.py

```diff
@@ -49,58 +51,69 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}()"
 
 
+@dataclass(frozen=True)
 class NoneWriter(Writer):
-    def write(self, _: BinaryEncoder, __: Any) -> None:
-        pass
+    def write(self, encoder: BinaryEncoder, __: Any) -> None:
+        encoder.write_int(0)
```

Comment:
I don't think this is correct. Avro encodes null as the null type, which has no byte representation. The only way to encode null is to select it as a branch of a union. But `null` in an Avro union can be anywhere.

I think the previous version was correct, where the NoneWriter skipped writing any bytes. That corresponds to writing a `null` value in Avro, where the containing option writer actually selects the null branch of the union.

Reply:
You're right here. The `NoneWriter` can go completely. We don't have to invoke the writer if we skip over an entry in the data schema.